### PR TITLE
more detailed deactivation

### PR DESCRIPTION
**Pull request by roosnic1**
WPC-337. This PR improves the deactivation field of a user subscription. It also introduces a check in the `chargeOpenInvoices` and the `sendReminderForInvoices` methods to cancel the invoice and deactivate the subscription if it has been tried to many times.

### CODE REVIEW

File: packages/api/src/memberContext.ts [hunk 1]
```diff
@@ -120,12 +133,18 @@ export class MemberContext implements MemberContext {
...
+  invoiceReminderMaxTries: number
+
   constructor(props: MemberContextProps) {
     this.dbAdapter = props.dbAdapter
     this.loaders = props.loaders
     this.paymentProviders = props.paymentProviders
 
     this.mailContext = props.mailContext
+
+    this.invoiceReminderFrequencyInDays = parseInt(process.env.INVOICE_REMINDER_FREQ ?? '') ?? 3
```
Comment (bishoymelek) [inline packages/api/src/memberContext.ts:146]: Should we add these env keys to .env.example?

File: packages/api/src/memberContext.ts [hunk 2]
```diff
@@ -585,22 +648,45 @@ export class MemberContext implements MemberContext {
...
-      if (
-        invoice.sentReminderAt &&
-        new Date(
+      if (!invoice.userID) {
+        logger('memberContext').warn('invoice %s does not have an user ID', invoice.id)
+        continue
+      }
+
+      if (invoice.sentReminderAt) {
+        const SHOULD_SEND_REMINDER_AGAIN_DATE = new Date(
```
Comment (bishoymelek) [inline packages/api/src/memberContext.ts:657]: Don't you think this following part duplicated? as in line 452. Can we have a function for this or?

**Review by camilleory — DISMISSED** [review]

File: packages/api/src/memberContext.ts [hunk 3]
```diff
@@ -585,22 +648,45 @@ export class MemberContext implements MemberContext {
     }
 
     for (const invoice of invoices.nodes) {
-      if (
-        invoice.sentReminderAt &&
-        new Date(
+      if (!invoice.userID) {
+        logger('memberContext').warn('invoice %s does not have an user ID', invoice.id)
```
Comment (penk85) [inline packages/api/src/memberContext.ts:652]: typo: "a user ID"

File: packages/editor/src/client/locales/en.json [hunk 4]
```diff
@@ -856,6 +856,32 @@
       "autoRenew": "Auto Renew",
       "autoRenewDescription": "Renew the subscription automatically",
       "createTitle": "Create Subscription",
+      "deactivation": {
+        "isDeactivated": "The subscription is deactivated since {{date, ccc PP p}} and needs to be reactivated before editing.",
+        "willBeDeactivated": "The subscription will be deactivated on {{date, ccc PP p}}",
+        "modalTitel": {
```
Comment (penk85) [inline packages/editor/src/client/locales/en.json:862]: Title

File: packages/editor/src/client/locales/en.json [hunk 5]
```diff
@@ -856,6 +856,32 @@
...
+          "deactivated": "Do you want to re-activate the subscription of \"{{userName}}\" with the email address \"{{userEmail}}\"?",
+          "activated": "Do you want to deactivate the subscription of \"{{userName}}\" with the email address \"{{userEmail}}\"?"
+        },
+        "title": {
+          "deactivated": "Reactivate subscription",
+          "activated": "Deactivate subscription"
+        },
+        "action": {
+          "deactivated": "Reactivate",
+          "activated": "Deactivates"
```
Comment (penk85) [inline packages/editor/src/client/locales/en.json:876]: This meant to be deactivates? (with s)

File: packages/api/src/memberContext.ts (hunk 1 above)
Comment (roosnic1) [reply packages/api/src/memberContext.ts:146]: No. We only add envs that are needed to run we.publish to `.env.example`

**Review by penk85 — APPROVED** [review]

**Review by camilleory — APPROVED** [review]

**Review by bishoymelek — APPROVED** [review]

**Pull request merged**